packages/db/src/lib/granule.ts

```diff
@@ -354,3 +357,39 @@ export const getGranulesByGranuleId = async (
     .where({ granule_id: granuleId });
   return records;
 };
+
+/**
+ * Change a granules' PG record and its files' PG record based on collection move
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+ * @param {Array<ApiGranule>} [granules] - updated ApiGranule records
+ * @returns {Promise<void>}
+ */
+export const updateGranuleAndFiles = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  granules: Array<ApiGranule>
+):Promise<void> => {
+  const {
+    granules: granulesTable,
+    files: filesTable,
+  } = TableNames;
+  await Promise.all(granules.map(async (granule) => {
```

Comment:
Question: If the map of granules is 10k, is it an acceptable usage scenario that when it fails on granule 1001 for <reasons>  that 1k granules were moved, and 9k failed, or do we want it to move 9,999 of them and fail the one with a metadata/connection/whatever issue?         Apologies if that should be obvious, I may be lacking context.